QUICKSTART-VMWARE.md

```diff
@@ -0,0 +1,248 @@
+    }
+  ],
+  "MarkAsTemplate": false,
+  "PowerOn": false,
+  "InjectOvfEnv": false,
+  "WaitForIP": false,
+  "Name": null
+}
+```
+
+You must enter valid values for the `Network` and `Name`.
+`Network` must correspond to the network in which you intend to deploy this VM.
+`Name` can be anything you want, it is the name of the VM that will be created.
+
+Once you've edited the file, you can use it to upload the OVA!
+```
+govc import.ova -options=bottlerocket_spec.json <bottlerocket...ova>
+```
+
+> Note: `govc import.ova` also takes a `-name=` argument that allows you to use the same spec file but override the VM name on future uploads.
+> For example: `govc import.ova -options=bottlerocket_spec.json -name="BottlerocketIsNeat" <bottlerocket...ova>`
+
+### Creating a template
+
+If you intend to run many identical VMs, you can mark the OVA you just uploaded as a template.
+You **must** to do this before you boot and configure Bottlerocket, since user data configuration will be unique between nodes!
+You can think of a template as a "golden" image, allowing you to create many VMs without affecting the "golden" image.
+
+The following command marks the VM as a template:
+```
+govc vm.markastemplate <YOUR VM NAME>
+```
+
+## Gathering cluster info
+
+This section will help you gather cluster information needed to configure Bottlerocket via user data.
+
+#### API Server
+This is the address (including port) of the control plane.
+You can run the following to get it:
+```
+ kubectl --kubeconfig=<your kubeconfig> cluster-info
+```
+The value you need will be found after `Kubernetes control plane is running at:` and should start with `https://` and end with a port.
+
+#### Cluster DNS IP
+This is the IP address of the DNS pod/service.
+```
+kubectl --kubeconfig=<your kubeconfig> -n kube-system get svc -l k8s-app=kube-dns
+```
+The value you need will be found under `CLUSTER-IP` and is an IP address.
+
+#### Bootstrap token
+Nodes require a token to establish trust between the node and the control plane.
+Tokens must be [used within 24 hours](https://kubernetes.io/docs/reference/setup-tools/kubeadm/kubeadm-token/), but once the node has booted and registered with the cluster, it isn't used again.
+```
+kubeadm --kubeconfig <your kubeconfig> token create
+```
+
+#### Cluster Certificate
+This is the base64 encoded cluster certificate authority.
+```
+kubectl --kubeconfig=<your kubeconfig> config view --raw -o json | jq -r '.clusters[0].cluster."certificate-authority-data"'
+```
+
+## Configuring Bottlerocket
+
+In order to join Bottlerocket to your cluster, it must be configured via user data.
+There are multiple methods of passing user data to Bottlerocket in VMware; we will demonstrate all of them.
+
+Create a file called `userdata.toml` and populate it with the values you just retrieved.
+```
+[settings.kubernetes]
+api-server = "<value here>"
+cluster-dns-ip = "<value here>"
+bootstrap-token = "<value here>"
+cluster-certificate = "<value here>"
```

Comment:
We can make this a copy/paste command, assuming they've followed along with other copy/paste commands and set the expected environment variables.

```suggestion
cat <<EOF > userdata.toml
[settings.kubernetes]
api-server = "${API_SERVER:-<API_SERVER>}"
cluster-dns-ip = "${CLUSTER_DNS_IP:-<CLUSTER_DNS_IP>}"
bootstrap-token = "${BOOTSTRAP_TOKEN:-<BOOTSTRAP_TOKEN>}"
cluster-certificate = "${CLUSTER_CERTIFICATE:-<CLUSTER_CERTIFICATE>}"
EOF
```